src/Compilers/VisualBasic/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.vb

```diff
@@ -0,0 +1,303 @@
+
+        Public Shared Function TryCreate(method As MethodSymbol, methodBody As BoundStatement, factory As SyntheticBoundNodeFactory, diagnostics As DiagnosticBag, debugDocumentProvider As DebugDocumentProvider, previous As Instrumenter) As DynamicAnalysisInjector
+            Dim createPayload As MethodSymbol = GetCreatePayload(factory.Compilation, methodBody.Syntax, diagnostics)
+
+            ' Do not instrument the instrumentation helpers if they are part of the current compilation (which occurs only during testing). GetCreatePayload will fail with an infinite recursion if it Is instrumented.
+            If createPayload IsNot Nothing AndAlso Not method.IsImplicitlyDeclared AndAlso Not method.Equals(createPayload) Then
+                Return New DynamicAnalysisInjector(method, methodBody, factory, createPayload, diagnostics, debugDocumentProvider, previous)
+            End If
+
+            Return Nothing
+        End Function
+
+        Public ReadOnly Property DynamicAnalysisSpans As ImmutableArray(Of SourceSpan)
+            Get
+                Return _dynamicAnalysisSpans
+            End Get
+        End Property
+
+        Private Sub New(method As MethodSymbol, methodBody As BoundStatement, factory As SyntheticBoundNodeFactory, createPayload As MethodSymbol, diagnostics As DiagnosticBag, debugDocumentProvider As DebugDocumentProvider, previous As Instrumenter)
+            MyBase.New(previous)
+            _createPayload = createPayload
+            _method = method
+            _methodBody = methodBody
+            _spansBuilder = ArrayBuilder(Of SourceSpan).GetInstance()
+            Dim payloadElementType As TypeSymbol = factory.SpecialType(SpecialType.System_Boolean)
+            _payloadType = ArrayTypeSymbol.CreateVBArray(payloadElementType, ImmutableArray(Of CustomModifier).Empty, 1, factory.Compilation.Assembly)
+            _methodPayload = factory.SynthesizedLocal(_payloadType, kind:=SynthesizedLocalKind.InstrumentationPayload, syntax:=methodBody.Syntax)
+            _diagnostics = diagnostics
+            _debugDocumentProvider = debugDocumentProvider
+            _methodHasExplicitBlock = MethodHasExplicitBlock(method)
+            _factory = factory
+
+            ' The first point indicates entry into the method And has the span of the method definition.
+            _methodEntryInstrumentation = AddAnalysisPoint(MethodDeclarationIfAvailable(methodBody.Syntax), factory)
+        End Sub
+
+        Public Overrides Function CreateBlockPrologue(original As BoundBlock, ByRef synthesizedLocal As LocalSymbol) As BoundStatement
+            Dim previousPrologue As BoundStatement = MyBase.CreateBlockPrologue(original, synthesizedLocal)
+
+            If _methodBody Is original Then
+                _dynamicAnalysisSpans = _spansBuilder.ToImmutableAndFree()
+                ' In the future there will be multiple analysis kinds.
+                Const analysisKind As Integer = 0
```

Comment:
I understand that you only have one `analysisKind` at the moment and you plan to have more in the future. It would be clearer what this concept means if you had an enum and gave this kind a name. 
Also, what other kinds might there be in the future?